src/utils/localization/fetchLocale.ts

```diff
@@ -0,0 +1,34 @@
+import { locales } from "../locale";
+
+/**
+ * Get supported locale code from raw user input
+ * Exported for testing purposes.
+ *
+ * @param lang
+ * @private
+ * @returns {string}
+ */
+export function getSupportedLocale(lang = ""): string {
```

Comment:
Please extract this in a way it can be shared between the date-picker utils and here.